salt/modules/file.py

```diff
@@ -2093,14 +2093,10 @@ def replace(path,
                         # Content was found, so set found.
                         found = True
 
-                # Keep track of show_changes here, in case the file isn't
-                # modified
-                if show_changes or append_if_not_found or \
-                   prepend_if_not_found:
-                    orig_file = r_data.read(filesize).splitlines(True) \
-                        if isinstance(r_data, mmap.mmap) \
-                        else r_data.splitlines(True)
-                    new_file = result.splitlines(True)
+                orig_file = r_data.read(filesize).splitlines(True) \
```

Comment:
The reason this was gated was because this doubles the memory requirement by keeping the file in two objects simultaneously. Might want to rethink the approach, or at least revisit the docs to mention the change (and remove the caveat on `show_changes`).